From 01d0ec72bd220c41d54e10ebafa503dd7951cb44 Mon Sep 17 00:00:00 2001
From: Jan Nieuwenhuizen <janneke@gnu.org>
Date: Tue, 1 Mar 2011 13:56:51 +0100
Subject: [PATCH] Midi2ly: use SEQUENCE_TRACK_NAME to set Staff.instrumentName.

---
 scripts/midi2ly.py |    4 ++++
 1 files changed, 4 insertions(+), 0 deletions(-)

diff --git a/scripts/midi2ly.py b/scripts/midi2ly.py
index a32d2aa..af47460 100644
--- a/scripts/midi2ly.py
+++ b/scripts/midi2ly.py
@@ -405,6 +405,10 @@ class Text:
                 or d.compare (reference_note.duration)):
                 s = s + Duration (self.clocks).dump ()
             s = s + ' '
+        elif self.text and self.type == midi.SEQUENCE_TRACK_NAME:
+            text = self.text.replace ('(MIDI)', '').strip ()
+            if text:
+                s = '\n  \\set Staff.instrumentName = "%(text)s"\n  ' % locals ()
         else:
             s = '\n  % [' + self.text_types[self.type] + '] ' + self.text + '\n  '
         return s
-- 
1.7.1

