From 1ba780c6f36cbd4b0adf7a21d7acbb2c6075817a Mon Sep 17 00:00:00 2001
From: Jan Nieuwenhuizen <janneke@gnu.org>
Date: Fri, 10 Dec 2010 13:06:59 +0100
Subject: [PATCH] [MINGW32]: Prepend cwd to PATH.  Fixes invoking as "lilypond.exe".

---
 lily/relocate.cc |    9 +++++----
 1 files changed, 5 insertions(+), 4 deletions(-)

diff --git a/lily/relocate.cc b/lily/relocate.cc
index 698f06a..178e7d9 100644
--- a/lily/relocate.cc
+++ b/lily/relocate.cc
@@ -201,17 +201,18 @@ setup_paths (char const *argv0_ptr)
 	      File_path path;
 	      path.parse_path (getenv ("PATH"));
 
-	      if (be_verbose_global)
-		warning (_f ("Relocation: from PATH=%s\nargv0=%s",
-			     path.to_string ().c_str (), argv0_ptr));
-
 #ifndef __MINGW32__
 	      argv0_abs = path.find (argv0_filename.to_string ());
 #else /* __MINGW32__ */
+	      path.prepend (get_working_directory ());
 	      char const *ext[] = {"exe", "", 0 };
 	      argv0_abs = path.find (argv0_filename.to_string (), ext);
 #endif /* __MINGW32__ */
 
+	      if (be_verbose_global)
+		warning (_f ("Relocation: from PATH=%s\nargv0=%s",
+			     path.to_string ().c_str (), argv0_ptr));
+
 	      if (argv0_abs.empty ())
 		programming_error ("cannot find absolute argv0");
 	    }
-- 
1.7.1

