diff -pu /home/janneke/vc/gub/target/cygwin/src/cross/gcc-4.3.4/libiberty/strsignal.c\~ /home/janneke/vc/gub/target/cygwin/src/cross/gcc-4.3.4/libiberty/strsignal.c
--- gcc-4.3.4/libiberty/strsignal.c~	2007-01-31 00:12:29.000000000 +0100
+++ gcc-4.3.4/libiberty/strsignal.c	2010-02-23 11:26:30.067355496 +0100
@@ -403,6 +403,12 @@ call to @code{strsignal}.
 */
 
 #ifndef HAVE_STRSIGNAL
+#ifdef __CYGWIN__
+#define HAVE_STRSIGNAL 1
+#endif /* __CYGWIN */
+#endif /* HAVE_STRSIGNAL */
+
+#ifndef HAVE_STRSIGNAL
 
 const char *
 strsignal (int signo)

Diff finished.  Tue Feb 23 11:26:58 2010
