--- a/base/gp_mswin.c	2014-11-29 16:00:46.394874900 +0900
+++ b/base/gp_mswin.c	2014-11-29 16:01:09.262874900 +0900
@@ -716,22 +716,3 @@
 #endif
 }
 
-/* -------------------------  _snprintf -----------------------------*/
-
-/* Microsoft Visual C++ 2005  doesn't properly define snprintf,
-   which is defined in the C standard ISO/IEC 9899:1999 (E) */
-
-int snprintf(char *buffer, size_t count, const char *format, ...)
-{
-    if (count > 0) {
-	va_list args;
-	int n; 
-
-	va_start(args, format);
-	n = _vsnprintf(buffer, count, format, args);
-	buffer[count - 1] = 0;
-	va_end(args);
-	return n;
-    } else
-	return 0;
-}
