--- ghostscript-8.70/base/unix-dll.mak.orig	2010-05-31 13:02:31.866745351 +0200
+++ ghostscript-8.70/base/unix-dll.mak	2010-05-31 13:03:08.779719110 +0200
@@ -79,11 +79,13 @@ $(GS_SO_MAJOR): $(GS_SO_MAJOR_MINOR)
 # Build the small Ghostscript loaders, with Gtk+ and without
 
 $(GSSOC_XE): $(GS_SO) $(PSSRC)$(SOC_LOADER)
+	-ln -f $(GS_SO_MAJOR_MINOR) $(BINDIR)/gs.dll
-	$(GLCC) -g -o $(GSSOC_XE) $(PSSRC)dxmainc.c -L$(BINDIR) -l$(GS)
+	$(GLCC) -g -o $(GSSOC_XE) $(PSSRC)dxmainc.c -L$(BINDIR) $(XLDFLAGS) -lgs
 
 $(GSSOX_XE): $(GS_SO) $(PSSRC)$(SOC_LOADER)
+	-ln -f $(GS_SO_MAJOR_MINOR) $(BINDIR)/gs.dll
 	$(GLCC) -g $(SOC_CFLAGS) -o $(GSSOX_XE) $(PSSRC)$(SOC_LOADER) \
-	-L$(BINDIR) -l$(GS) $(SOC_LIBS)
+	-L$(BINDIR) $(XLDFLAGS) -lgs $(SOC_LIBS)
 
 # ------------------------- Recursive make targets ------------------------- #
 
