diff -purN inkscape-trunk/src/color-profile.cpp.orig inkscape-trunk/src/color-profile.cpp
--- inkscape-trunk/src/color-profile.cpp.orig	2010-06-03 13:54:10.995489100 +0200
+++ inkscape-trunk/src/color-profile.cpp	2010-06-03 13:54:21.216685738 +0200
@@ -34,7 +34,7 @@
 #include "dom/util/digest.h"
 
 #ifdef WIN32
-#include <Icm.h>
+#include <icm.h>
 #endif // WIN32
 
 using Inkscape::ColorProfile;
--- inkscape-trunk-20605/src/ui/dialog/filedialogimpl-win32.cpp~	2009-01-30 13:37:43.000000000 +0100
+++ inkscape-trunk-20605/src/ui/dialog/filedialogimpl-win32.cpp	2009-02-06 16:34:10.000000000 +0100
@@ -1321,10 +1321,10 @@ void FileOpenDialogImplWin32::render_pre
     if(_preview_bitmap_image)    // Is the image a pixbuf?
     {
         // Set the transformation
-        const Matrix matrix = {
+        const Matrix matrix (
             scaleFactor, 0,
             0, scaleFactor,
-            svgX, svgY };
+            svgX, svgY );
         context->set_matrix (matrix);
 
         // Render the image
--- inkscape-trunk/src/dialogs/export.cpp~	2010-06-03 10:48:54.000000000 +0200
+++ inkscape-trunk/src/dialogs/export.cpp	2010-06-03 14:05:34.054237582 +0200
@@ -62,7 +62,7 @@
 
 #ifdef WIN32
 #include <windows.h>
-#include <COMMDLG.h>
+#include <commdlg.h>
 #include <gdk/gdkwin32.h>
 #endif
 
--- inkscape-trunk/configure.ac~	2010-06-03 14:06:14.334256081 +0200
+++ inkscape-trunk/configure.ac	2010-06-03 15:11:42.944239713 +0200
@@ -261,6 +261,7 @@ case "$host" in
   *-*-mingw*)
     platform_win32=yes
     WIN32_CFLAGS="-mms-bitfields -DLIBXML_STATIC"
+    W32_INKSCAPE_LIBS="-lmscms"
     ;;
   *)
     platform_win32=no
@@ -270,6 +270,7 @@ esac
 AC_MSG_RESULT([$platform_win32])
 AM_CONDITIONAL(PLATFORM_WIN32, test "$platform_win32" = "yes")
 AC_SUBST(WIN32_CFLAGS)
+AC_SUBST(W32_INKSCAPE_LIBS)
 
 AC_MSG_CHECKING([for Solaris platform])
 case "$host" in
--- inkscape-trunk/src/Makefile.am.orig	2010-06-03 16:53:37.974239005 +0200
+++ inkscape-trunk/src/Makefile.am	2010-06-03 16:54:18.615490866 +0200
@@ -40,6 +40,7 @@ all_libs =			\
 	$(FREETYPE_LIBS)	\
 	$(kdeldadd)		\
 	$(win32ldflags)		\
+	$(W32_INKSCAPE_LIBS)	\
 	$(CARBON_LDFLAGS)	\
 	$(PERL_LIBS)		\
 	$(PYTHON_LIBS)		\
