--- pango-1.36.8/tests/Makefile.in.org	2014-09-23 03:30:19.000000000 +0900
+++ pango-1.36.8/tests/Makefile.in	2015-08-30 13:36:48.925365600 +0900
@@ -84,8 +84,8 @@
 @HAVE_WIN32_TRUE@am__append_3 = -DHAVE_WIN32
 @CROSS_COMPILING_FALSE@TESTS = $(check_PROGRAMS)
 check_PROGRAMS = testboundaries$(EXEEXT) testboundaries_ucd$(EXEEXT) \
-	testcolor$(EXEEXT) testscript$(EXEEXT) markup-parse$(EXEEXT) \
-	test-layout$(EXEEXT) test-font$(EXEEXT) $(am__EXEEXT_1) \
+	testcolor$(EXEEXT) testscript$(EXEEXT) \
+	$(am__EXEEXT_1) \
 	$(am__EXEEXT_2) $(am__EXEEXT_3)
 @HAVE_CAIRO_TRUE@am__append_4 = testiter test-pangocairo-threads
 @HAVE_FREETYPE_TRUE@am__append_5 = test-ot-tags
