--- python-2.4.5/Modules/Setup.dist~	2012-08-26 17:54:57.188647513 +0200
+++ python-2.4.5/Modules/Setup.dist	2012-08-26 18:10:03.592290524 +0200
@@ -377,7 +377,7 @@
 #
 # First, look at Setup.config; configure may have set this for you.
 
-#dbm dbmmodule.c 	# dbm(3) may require -lndbm or similar
+dbm dbmmodule.c -DHAVE_BERKDB_H -DDB_DBM_HSEARCH -ldb	# dbm(3) may require -lndbm or similar
 
 # Anthony Baxter's gdbm module.  GNU dbm(3) will require -lgdbm:
 #
