--- python-2.4.5/Modules/Setup.dist~	2005-12-27 18:37:16.000000000 +0100
+++ python-2.4.5/Modules/Setup.dist	2012-08-26 15:53:56.245136116 +0200
@@ -212,7 +212,7 @@ GLHACK=-Dclear=__GLclear
 #
 # First, look at Setup.config; configure may have set this for you.
 
-#crypt cryptmodule.c # -lcrypt	# crypt(3); needs -lcrypt on some systems
+crypt cryptmodule.c # -lcrypt	# crypt(3); needs -lcrypt on some systems
 
 
 # Some more UNIX dependent modules -- off by default, since these
